File: src/app/Model/product.ts
export class Product {
    public productId: number;
    public productName: string;
    public productTexture: string;
    public productGrammage: number;
    public productColor: string;
    public constructor(id: number, name: string, texture: string, grammage: number, color: string) {
        this.productId = id;
        this.productName = name;
        this.productTexture = texture;
        this.productGrammage = grammage;
        this.productColor = color;
    }
}

File: API/ProductApi/ViewModels/Product.cs
using System.ComponentModel.DataAnnotations;

namespace ProductAPI.Models
{
  public class Product
  {
    public int productId { get; set; }

    [Required]
    [MaxLength(100)]
    public string productName { get; set; }

    [Required]
    [MaxLength(50)]
    public string productTexture { get; set; }

    [Required]
    [Range(10, 1000)]
    public int productGrammage { get; set; }

    [MaxLength(25)]
    public string productColor { get; set; }
  }
}
